File: Controllers/PaymentController.js
// PaymentController.js
// Replace `payments` array with backend API calls later

let payments = [];

const PaymentController = {

    getAll() {
        return payments;
    },

    getByMember(memberID) {
        return payments.filter(p => p.memberID === memberID);
    },

    process(memberID, amount) {
        const paymentID = payments.length > 0 ? payments[payments.length - 1].paymentID + 1 : 1;
        const paymentDate = new Date().toLocaleString();
        payments.push({ paymentID, memberID, amount, paymentDate });
        return true;
    },

    remove(paymentID) {
        const index = payments.findIndex(p => p.paymentID === paymentID);
        if (index === -1) {
            alert("Payment not found!");
            return false;
        }
        payments.splice(index, 1);
        return true;
    }
};

File: Controllers/PaymentController.cpp
#include "PaymentController.h"
#include <iostream>
using namespace std;

PaymentController::PaymentController(Database& db) : db(db) {}

void PaymentController::getAll() {
    cout << "\n--- ALL PAYMENTS ---\n";
    db.loadPayments();
}

void PaymentController::getByMember(int memberID) {
    cout << "\n--- PAYMENTS FOR MEMBER ID: " << memberID << " ---\n";
    db.loadPaymentsByMember(memberID);
}

void PaymentController::process(int memberID, double amount) {
    Payment p(0, amount);
    db.savePayment(memberID, amount);
    cout << "✔ Payment of $" << amount << " processed for Member ID " << memberID << ".\n";
}

void PaymentController::remove(int paymentID) {
    db.deletePayment(paymentID);
    cout << "✔ Payment ID " << paymentID << " deleted.\n";
}